packages/mask/src/plugins/Referral/SNSAdaptor/CreatedFarms.tsx

```diff
@@ -0,0 +1,142 @@
+
+    return (
+        <>
+            {farms.map((farm) => {
+                return (
+                    <AccordionFarm
+                        key={farm.farmHash}
+                        rewardToken={farm.rewardToken}
+                        referredToken={farm.referredToken}
+                        totalValue={Number.parseFloat(farm?.totalFarmRewards?.toFixed(5) ?? '0')}>
+                        <Box display="flex" justifyContent="flex-end">
+                            <Button variant="text" disabled classes={{ disabled: classes.viewStatsDisabled }}>
+                                {t.view_stats()}
+                            </Button>
+                            <Button
+                                disabled
+                                variant="contained"
+                                size="medium"
+                                className={classes.buttonWithdraw}
+                                onClick={() => console.log('request to withdraw')}>
+                                {t.request_to_withdraw()}
+                            </Button>
+                            <Button
+                                variant="contained"
+                                size="medium"
+                                onClick={() => {
+                                    onAdjustRewardButtonClick(farm)
+                                }}>
+                                {t.adjust_rewards()}
+                            </Button>
+                        </Box>
+                    </AccordionFarm>
+                )
+            })}
+        </>
+    )
+}
+
+export function CreatedFarms(props: PageInterface) {
+    const t = useI18N()
+    const { classes: sharedClasses } = useSharedStyles()
+    const { classes: myFarmsClasses } = useMyFarmsStyles()
+    const currentChainId = useChainId(NetworkPluginID.PLUGIN_EVM)
+    const requiredChainId = getRequiredChainId(currentChainId)
+    const account = useAccount(NetworkPluginID.PLUGIN_EVM)
+
+    const {
+        value: farms = EMPTY_LIST,
+        loading,
+        error,
+    } = useAsync(
+        async () => (account ? ReferralRPC.getAccountFarms(account, currentChainId) : []),
+        [currentChainId, account],
+    )
+
+    const onAdjustRewardButtonClick = (farm: FarmDetailed) => {
+        props.continue(PagesType.CREATE_FARM, PagesType.ADJUST_REWARDS, t.adjust_rewards(), {
+            adjustFarmDialog: {
+                farm,
+                rewardToken: farm.rewardToken,
+                referredToken: farm.referredToken,
+                continue: () => {},
+            },
+        })
+    }
+
+    if (currentChainId !== requiredChainId) {
+        return (
+            <ChainBoundary expectedChainId={requiredChainId} expectedPluginID={NetworkPluginID.PLUGIN_EVM}>
+                <WalletConnectedBoundary offChain />
+            </ChainBoundary>
+        )
+    }
```

Comment:
If the user selected a non-EVM network, this condition will not be fulfilled. Try this:

```tsx
return (
    <ChainBoundary>
        <WalletConnectedBoundary>
            <UI />
        </WalletConnectedBoundary>
    </ChainBoundary>
)
```

Reply:
Thanks, good point, done.